### DISCUSSION TITLE
Avoid stroke overlap when widgets in close proximity

### DISCUSSION THREAD

**Discussion by Beinsezii** [emilk/egui#2032] [Q&A]
I've got some Buttons in close proximity on a Grid that show a 1.0 Stroke on hover, and the stroke only displays over widgets added before the hovered in question. I'm trying to specifically make a 2-tone UI, and the current setup is

 * Item spacing 0.0
 * `Stroke::none()` on all states except `hovered`, where it's `1.0`

If I set the spacing to something like 2.0 to allow for full stroke width, the Grid's bg_fill being the same color shows through. Also setting `inactive`'s stroke to 2.0 black covers it up, but the `hovered` stroke still clips, resulting in a full size stroke on lower widgets and a half size stroke on higher widgets. Alternatively, setting a tiny golden stroke with a black Grid fill on 2.0 spacing results in an `airgapped` double border around everything. It's consistent at least but it looks funky.

I noticed there's a property `Visuals.clip_rect_margin` which sounds like it'd be useful, but it seems to have no effect.

Attached image with 0.0 spacing and 1.0 stroke while hovering demoing problem when 3 is hovered. Stroke shows up over 1, but is covered by 4 which gets added after 3.

![butts](https://user-images.githubusercontent.com/39478211/189282167-4555d5f8-f7d3-4013-9a8c-e9f486902d27.png)

**TL;DR:** buttons close proximity, want stroke on `hovered`, stroke clipped by `inactive` buttons. Wondering if I could make the stroke draw inwards instead of outwards or something.
